File: 1-twosum/solution.js
export default twoSum

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
function twoSum(nums, target) {
  if (!nums) return

  const complements = new Map()
  for (let i = 0; i < nums.length; i++) {
    if (complements.has(nums[i])) return [complements.get(nums[i]), i]
    complements.set(target - nums[i], i)
  }
}

File: 1-twosum/solution.go
package solutionTwoSum

func TwoSum(nums []int, target int) []int {
	if nums == nil {
		return nil
	}

	complements := make(map[int]int)
	for i, num := range nums {
		if seen_idx, hasSeen := complements[num]; hasSeen {
			return []int{seen_idx, i}
		}
		complements[target-num] = i
	}

	return nil
}
